src/OpenSage.Game/Logic/Object/Upgrade/MaxHealthUpgrade.cs

```diff
@@ -1,7 +1,41 @@
 ﻿using OpenSage.Data.Ini;
+using OpenSage.Mathematics.FixedMath;
 
 namespace OpenSage.Logic.Object
 {
+    public sealed class MaxHealthUpgrade : UpgradeModule
+    {
+        private readonly GameObject _gameObject;
+
+        internal MaxHealthUpgrade(GameObject gameObject, MaxHealthUpgradeModuleData moduleData) : base(moduleData)
+        {
+            _gameObject = gameObject;
+        }
+
+        internal override void OnTrigger(BehaviorUpdateContext context, bool triggered)
+        {
+            if (triggered)
+            {
+                var maxHealthUpgrade = _moduleData as MaxHealthUpgradeModuleData;
+                _gameObject.Body.MaxHealth += (Fix64) maxHealthUpgrade.AddMaxHealth;
+
+                switch (maxHealthUpgrade.ChangeType)
+                {
+                    case MaxHealthChangeType.PreserveRatio:
+                        Fix64 ratio = _gameObject.Body.Health / _gameObject.Body.MaxHealth;
```

Comment:
I'd have thought this should be `Fix64 ratio = _gameObject.Body.Health / <MaxHealth before it was changed just above>;`

Reply:
Good catch! I'll put the maxHealth assignment below the switch statement

Reply:
Fixed this